esphome/components/deep_sleep/deep_sleep_component.h

```diff
@@ -213,5 +219,11 @@ template<typename... Ts> class AllowDeepSleepAction : public Action<Ts...>, publ
   void play(Ts... x) override { this->parent_->allow_deep_sleep(); }
 };
 
+class DeepSleepTrigger : public Trigger<>, public Parented<DeepSleepComponent> {
```

Comment:
Not required as it is not being stored for later use.
```suggestion
class DeepSleepTrigger : public Trigger<> {
```

Reply:
removed it